### PR TITLE
Fix async decode

### PR DESCRIPTION
**Pull request by harmony-ek**

### CODE REVIEW

File: pkg/internal/readyblockchan/readyblockchan.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+import (
+	"fmt"
+	"sync"
+)
+
+type AlreadyAdded chan<- uint8
+
+func (e AlreadyAdded) Error() string {
+	return fmt.Sprintf("ready-block channel %+v already added")
```
Comment (chaosma) [inline pkg/internal/readyblockchan/readyblockchan.go]: It seems that %+v not work here.

File: pkg/internal/readyblockchan/readyblockchan.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+type NotFound chan<- uint8
+
+func (e NotFound) Error() string {
+	return fmt.Sprintf("ready-block channel %+v not found")
```
Comment (chaosma) [inline pkg/internal/readyblockchan/readyblockchan.go]: %+v not work here

**Review by chaosma — COMMENTED** [review]
Looks good. Some minor changes may needed

File: pkg/internal/readyblockchan/readyblockchan.go (hunk 1 above)
Comment (harmony-ek) [reply pkg/internal/readyblockchan/readyblockchan.go]: Good catch, will fix.

File: pkg/internal/readyblockchan/readyblockchan.go (hunk 2 above)
Comment (harmony-ek) [reply pkg/internal/readyblockchan/readyblockchan.go]: Good catch, will fix (2).

**Review by chaosma — APPROVED** [review]

**Pull request merged**